src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -85,6 +91,124 @@ class ResultsNeedFullReloadError(RuntimeError):
+    :ivar network_element_name:
+        The name of network where this data is associated.
+    :ivar position:
+        The element position in a pipe.
+    :ivar position_unit:
+        The position unit (m, Km, cm).
+    :ivar unit:
+        The unit of a dimensionless property (-, %)
+    :ivar qoi_index:
+        The index of a quantity of interest in the results file.
+    :ivar qoi_data_index:
+        The data index of a quantity of interest.
+    """
+
+    property_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    trend_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    category: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_name: str = attr.ib(validator=attr.validators.instance_of(str))
+    network_element_name: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    position: Optional[float] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(float))
+    )
+    position_unit: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    unit: str = attr.ib(validator=attr.validators.instance_of(str))
+    qoi_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+    qoi_data_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+
+
+@attr.s(slots=True, hash=False)
+class GlobalSensitivityAnalysisMetadata:
+    """
+    A class that hold the global sensitivity analysis metadata.
+    """
+
+    global_sensitivity_analysis_meta_data: Dict[
+        str, GlobalSensitivityAnalysisMetaInformation
+    ] = attr.ib(validator=attr.validators.instance_of(Dict))
+    result_directory: Path = attr.ib(validator=attr.validators.instance_of(Path))
+
+    @classmethod
+    def empty(cls, result_directory: Path) -> "GlobalSensitivityAnalysisMetadata":
+        return GlobalSensitivityAnalysisMetadata(
+            global_sensitivity_analysis_meta_data={}, result_directory=result_directory
+        )
+
+    @classmethod
+    def from_dict(
+        cls, data: dict[str, Any]
+    ) -> "GlobalSensitivityAnalysisMetaInformation":
+        return GlobalSensitivityAnalysisMetaInformation(
+            property_id=data["property_id"],
+            trend_id=data["trend_id"],
+            category=data["category"],
+            parametric_var_id=data["parametric_var_id"],
+            parametric_var_name=data["parametric_var_name"],
+            network_element_name=data["network_element_name"],
+            position=data["position"],
+            position_unit=data["position_unit"],
+            unit=data["unit"],
+            qoi_index=data["qoi_index"],
+            qoi_data_index=data["qoi_data_index"],
+        )
```

Comment:
```suggestion
    @classmethod
    def from_dict(
        cls, data: dict[str, Any]
    ) -> Self:
        return cls(
            property_id=data["property_id"],
            trend_id=data["trend_id"],
            category=data["category"],
            parametric_var_id=data["parametric_var_id"],
            parametric_var_name=data["parametric_var_name"],
            network_element_name=data["network_element_name"],
            position=data["position"],
            position_unit=data["position_unit"],
            unit=data["unit"],
            qoi_index=data["qoi_index"],
            qoi_data_index=data["qoi_data_index"],
        )
```

Reply:
https://docs.python.org/3/library/typing.html#typing.Self

> New in version 3.11.

We can't use that because alfasim-sdk supports python 3.8, 3.9, and 3.10, and I don't think there is a future for `Self`.

Reply:
It is importable from `typing_extensions` and supported by all those version, please let's use it.

Reply:
The ```from_dict``` method iis defined in ```GlobalSensitivityAnalysisMetadata``` but they return another type od data: ```GlobalSensitivityAnalysisMetaInformation```. So it could not be used.

Reply:
Ahh definitely, my bad.